palace/CMakeLists.txt

```diff
@@ -100,28 +58,58 @@ if(PALACE_WITH_OPENMP)
   find_package(OpenMP REQUIRED)
 endif()
 
-# Find PETSc and SLEPc
-include(PkgConfigHelpers)
-set(PETSC_TEST_DEPS MPI::MPI_CXX)
-if(PALACE_WITH_OPENMP)
-  list(APPEND PETSC_TEST_DEPS OpenMP::OpenMP_CXX)
+# Find nlohmann/json
+find_package(nlohmann_json REQUIRED CONFIG)
+message(STATUS "Found nlohmann/json: ${nlohmann_json_VERSION} in ${nlohmann_json_DIR}")
+
+# Find fmt
+find_package(fmt REQUIRED CONFIG)
+message(STATUS "Found fmt: ${fmt_VERSION} in ${fmt_DIR}")
+
+# Find Eigen
+find_package(Eigen3 REQUIRED CONFIG)
+message(STATUS "Found Eigen: ${Eigen3_VERSION} in ${Eigen3_DIR}")
+
+# Find MFEM
+if(NOT "${MFEM_DIR}" STREQUAL "")
+  set(MFEM_ROOT ${MFEM_DIR})
 endif()
-find_petsc_pkgconfig("${PETSC_TEST_DEPS}" PETSC_TARGET)
-if("${PETSC_TARGET}" STREQUAL "")
-  message(FATAL_ERROR "PETSc could not be found, be sure to set PETSC_DIR")
+find_package(MFEM REQUIRED CONFIG)
+message(STATUS "Found MFEM: ${MFEM_VERSION} in ${MFEM_DIR}")
+if(NOT MFEM_USE_MPI)
+  message(FATAL_ERROR "Build requires MFEM with MPI support")
 endif()
+# if(MFEM_CXX_FLAGS)
+#   # Pull compiler flags from MFEM for OpenMP and optimizations
+#   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${MFEM_CXX_FLAGS}")
+# endif()
```

Comment:
debris?

Reply:
Likewise, leaving this in case it becomes useful (this functionality is not new here).